--- contrib/sb-posix/posix-tests.lisp.orig	2016-01-05 07:18:26.000000000 +0000
+++ contrib/sb-posix/posix-tests.lisp	2016-01-05 07:21:27.000000000 +0000
@@ -103,6 +103,7 @@
       (sb-posix:syscall-errno c)))
   #.sb-posix::eexist)
 
+#-freebsd
 (deftest mkdir.error.2
   (handler-case
       (sb-posix:mkdir #-win32 "/" #+win32 "C:/" 0)
@@ -167,6 +168,7 @@
   #+win32
   #.sb-posix::einval)
 
+#-freebsd
 (deftest rmdir.error.3
   (handler-case
       (sb-posix:rmdir #-win32 "/" #+win32 (sb-ext:posix-getenv "windir"))
@@ -339,6 +341,7 @@
             (,mode (sb-posix::stat-mode ,stat)))
        ,@body)))
 
+#-freebsd
 (deftest stat-mode.1
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isreg mode))
@@ -354,11 +357,13 @@
     (sb-posix:s-ischr mode))
   nil)
 
+#-freebsd
 (deftest stat-mode.4
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isblk mode))
   nil)
 
+#-freebsd
 (deftest stat-mode.5
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isfifo mode))
@@ -443,7 +448,7 @@
                   sb-posix::o-nonblock)))
   t)
 
-#-(or hpux win32 netbsd) ; fix: cant handle c-vargs
+#-(or hpux win32 netbsd freebsd) ; fix: cant handle c-vargs
 (deftest fcntl.flock.1
     (locally (declare (sb-ext:muffle-conditions sb-ext:compiler-note))
       (let ((flock (make-instance 'sb-posix:flock
@@ -478,7 +483,7 @@
   42)
 
 
-#-(or win32 netbsd)
+#-(or win32 netbsd freebsd)
 (deftest fcntl.flock.2
     (locally (declare (sb-ext:muffle-conditions sb-ext:compiler-note))
       (let ((flock (make-instance 'sb-posix:flock
@@ -535,6 +540,7 @@
         (sb-posix:closedir dir))))
   nil)
 
+#-freebsd
 (deftest readdir.1
   (let ((dir (sb-posix:opendir "/")))
     (unwind-protect
@@ -708,6 +714,7 @@
     #.(concatenate 'string "/" (make-string 255 :initial-element #\a)))
 
   ;; The error tests are in the order of exposition from SUSv3.
+  #-freebsd
   (deftest readlink.error.1
       (let* ((subdir-pathname (merge-pathnames
                                (make-pathname
